plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/InfluxRecordCursor.java

```diff
@@ -0,0 +1,208 @@
+            for (String name : names) {
+                Object value = record.get(nameIdxMap.get(name));
+                alignedRecord.add(value);
+            }
+            alignedRecordSet.add(alignedRecord);
+        }
+
+        this.sourceDataIterator = alignedRecordSet.iterator();
+
+        try (ByteArrayOutputStream bo = new ByteArrayOutputStream(); ObjectOutputStream oo = new ObjectOutputStream(bo)) {
+            oo.writeObject(sourceData);
+            byte[] bytes = bo.toByteArray();
+            CountingInputStream input = new CountingInputStream(new ByteArrayInputStream(bytes));
+            totalBytes = input.getCount();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return totalBytes;
+    }
+
+    @Override
+    public long getReadTimeNanos()
+    {
+        return 0;
+    }
+
+    @Override
+    public Type getType(int field)
+    {
+        checkArgument(field < columnHandles.size(), "Invalid field index");
+        return columnHandles.get(field).getType();
+    }
+
+    @Override
+    public boolean advanceNextPosition()
+    {
+        if (!sourceDataIterator.hasNext()) {
+            return false;
+        }
+        fieldValues = sourceDataIterator.next();
+        return true;
+    }
+
+    private Object getFieldValue(int field)
+    {
+        checkState(fieldValues != null, "Cursor has not been advanced yet");
+        return fieldValues.get(field);
+    }
+
+    @Override
+    public boolean getBoolean(int field)
+    {
+        checkFieldType(field, BOOLEAN);
+        Object value = getFieldValue(field);
+        if (value instanceof Boolean) {
+            return (Boolean) value;
+        }
+        if (value instanceof Number number) {
+            return number.intValue() != 0;
+        }
+        return false;
+    }
+
+    @Override
+    public long getLong(int field)
+    {
+        checkFieldType(field, BIGINT);
+        Object value = getFieldValue(field);
+        if (value instanceof Number number) {
+            return number.longValue();
+        }
+        return 0;
+    }
+
+    @Override
+    public double getDouble(int field)
+    {
+        checkFieldType(field, DOUBLE);
+        Object value = getFieldValue(field);
+        if (value instanceof Number) {
+            return ((Number) value).doubleValue();
+        }
+        return 0;
+    }
+
+    @Override
+    public Slice getSlice(int field)
+    {
+        checkFieldType(field, createUnboundedVarcharType());
+        Object value = getFieldValue(field);
+        return utf8Slice((String) value);
+    }
+
+    @Override
+    public Object getObject(int field)
+    {
+        Type type = getType(field);
+        Object value = getFieldValue(field);
+        if (type == TIMESTAMP_NANOS) {
+            Instant utc = LocalDateTime.parse((String) value, ISO_DATE_TIME).atZone(ZoneId.of("UTC")).toInstant();
```

Comment:
How is timestamp defined in InfluxDB - do we need to map it to Timestamp with Time Zone

Reply:
It's a unix nanosecond timestamp stores in influxdb, but return a UTC-timestamp string when we use influx-client to query. https://docs.influxdata.com/influxdb/v1.8/write_protocols/line_protocol_reference/#data-types